packages/fiber/src/core/index.tsx

```diff
@@ -348,12 +347,19 @@ function Portal({
       if (injectState) {
         // Only the fields of "state" that do not differ from injectState
 
-        // Some props should be off-limits. Discarding off-limits props from being checked
-        let { size: _, viewport: __, internal: ___, performance: ____, ...otherState } = prevRootState
-        for (let key in otherState) {
+        const protectedKeys: Readonly<Record<string, string>> = {
```

Comment:
Bring back the protected keys as requested. It's now an object (it was an array) and also moved to outside the loop